src/engine/eip4844.md

```diff
@@ -0,0 +1,49 @@
+# EIP-4844 Engine API extension
+
+This is an extension specific to EIP-4844 to the `core` methods as defined in the [Engine API](./specification.md).
+This extension is backwards-compatible, but not part of the initial Engine API.
+
+## Structures
+
+### BlobsBundleV1
+
+The fields are encoded as follows:
+
+- `blockHash`: `DATA`, 32 Bytes
+- `kzgs`: `Array of DATA` - Array of `KZGCommitment` as defined in [EIP-4844](https://eips.ethereum.org/EIPS/eip-4844), 48 bytes each (`DATA`).
+- `blobs`: `Array of DATA` - Array of blobs, each blob is `FIELD_ELEMENTS_PER_BLOB * size_of(BLSFieldElement) = 4096 * 32 = 131072` bytes (`DATA`) representing a SSZ-encoded `Blob` as defined in [EIP-4844](https://eips.ethereum.org/EIPS/eip-4844)
+
+## Methods
+
+### engine_getBlobsBundleV1
+
+This method retrieves the blobs and their respective KZG commitments corresponding to the `versioned_hashes`
+included in the blob transactions of the referenced execution payload.
+
+This method may be combined with `engine_getPayloadV1` into a `engine_getPayloadV2` in a later stage of EIP-4844.
+The separation of concerns aims to minimize changes during the testing phase of the EIP.
```

Comment:
Not sure if it minimizes an overall complexity. IIUC, this method call should be synchronized with `getPayloadV1` call in the following way:
* CL calls `getBlobsBundleV1` and EL stops updating the payload
* CL calls `getPayloadV1` to obtain an execution payload and then should match the `blockHash` value

IMO, it makes sense to go straight with the `getPayloadV2` option. It looks like its implementation is copy&paste `getPayloadV1` + adding blobs and kzgs logic. I might be wrong tho

Reply:
This decision was based on implementer feedback from Prysm/Lighthouse devs working with this API. Long-term everyone likes `getPayloadV2` better, but short-term we want to avoid any overlap with the regular engine API.